keps/sig-cloud-provider/003-testgrid-e2e.md

```diff
@@ -0,0 +1,224 @@
+Done.
+```
+
+##### Testgrid Configuration
+
+Next thing you want to do is configure testgrid to read results from your GCS bucket. There are two [configuration](https://github.com/kubernetes/test-infra/tree/master/testgrid#configuration) steps required. One for your [test group](https://github.com/kubernetes/test-infra/tree/master/testgrid#test-groups) and one for your [dashboard](https://github.com/kubernetes/test-infra/tree/master/testgrid#dashboards).
+
+To add a test group update [config.yaml](https://github.com/kubernetes/test-infra/blob/master/testgrid/config.yaml) with something like the following:
+```
+test_groups:
+...
+...
+- name: cloud-provider-demo-e2e-conformance-release-v1.10
+  gcs_prefix: k8s-conformance-demo/cloud-provider-demo/e2e-conformance-release-v1.10
+```
+
+To add a link to your results in the testgrid dashboard, update [config.yaml](https://github.com/kubernetes/test-infra/blob/master/testgrid/config.yaml) with something like the following:
+```
+dashboards:
+...
+...
+- name: conformance-demo-cloud-provider
+  dashboard_tab:
+  - name: "Demo Cloud Provider, v1.10"
+    description: "Runs conformance tests for cloud provier demo on release v1.10"
+    test_group_name: cloud-provider-demo-e2e-conformance-release-v1.10
+```
+
+Once you've made the following changes, open a PR against the test-infra repo. Once your PR merges you should be able to view your results on https://k8s-testgrid.appspot.com/ which should be ready to be consumed by the necessary stakeholders (sig-release, sig-testing, etc).
+
+
+#### Examples
+
+Here are some more concrete examples of how other cloud providers are running conformance tests and uploading results to testgrid:
+* Open Stack
+    * [OpenLan zuul job for running/uploading testgrid results](https://github.com/theopenlab/openlab-zuul-jobs/tree/master/playbooks/cloud-provider-openstack-acceptance-test-e2e-conformance)
+    * [OpenStack testgrid config](https://github.com/kubernetes/test-infra/pull/7670)
+    * [OpenStack conformance tests dashboard](https://github.com/kubernetes/test-infra/pull/8154)
+
+
+### Risks and Mitigations
+
+#### Operational Overhead
+
+Operating CI/CD system to run conformance tests on a regular basis may incur extra work from every cloud provider. Though we anticipate the benefits of running conformance tests to outweight the operational overhead, in some cases it may not.
+
+Mitigation: TODO
+
+#### Misconfigured Tests
+
+There are various scenarios where cloud providers may mistakenly upload incorrect conformance tests results. One example being uploading results for the wrong Kubernetes version.
+
+Mitigation: TODO
+
+#### Flaky Tests
+
+Tests can fail for various reasons in any cloud environment and may raise false negatives for the release team.
```

Comment:
we expect some level of flaking to occur, but the release team will probably not move/keep jobs into the release blocking dashboards if they are very flaky. (they will still be in other dashboards of course). for the moment which jobs are in the release dashboard is ~up to the release team, and they may choose to ignore other dashboards.

Reply:
+1 on this. One of the major criteria in promoting a e2e test to conformance is it should flake <1% of the time. If for some reason this isnt met we will remove the test from conformance after investigation of rootcause and fix.

Reply:
Sounds good, I will update to use `kubetest`, I was in favour of Sonobuoy originally because it was easier to run since it doesn't require setting up a development environment but I don't think the tradeoff is worth it.